tests/tests/shader/data_builtins.rs

```diff
@@ -0,0 +1,127 @@
+use wgpu::{DownlevelFlags, Limits};
+
+use crate::shader::{shader_input_output_test, InputStorageType, ShaderTest};
+use wgpu_test::{gpu_test, GpuTestConfiguration, TestParameters};
+
+#[allow(non_snake_case)]
+fn create_unpack4xU8_test() -> Vec<ShaderTest> {
+    let mut tests = Vec::new();
+
+    let input: u32 = 0xAABBCCDD;
+    let output: [u32; 4] = [0xDD, 0xCC, 0xBB, 0xAA];
+    let unpack_u8 = ShaderTest::new(
+            format!("unpack4xU8({input:x?}) == {output:x?}"),
+            String::from("value: u32"),
+            String::from("let a = unpack4xU8(input.value); output[0] = a[0];output[1] = a[1];output[2] = a[2];output[3] = a[3];"),
```

Comment:
For readability let's make this multiple lines
```suggestion
            String::from("
                let a = unpack4xU8(input.value);
                output[0] = a[0];
                output[1] = a[1];
                output[2] = a[2];
                output[3] = a[3];
            "),
```

Reply:
Done!

Reply:
Weird quotes placement is enforced by rustfmt, so it isn't exactly as your suggested change, but I assume that's okay

Reply:
Yeah, that's fine.